src/android-verify.js

```diff
@@ -52,30 +69,56 @@ export const verify = async (data: Uint8Array) => {
       }
     }
     verifyAttestationCertChain(leafCert, intermediateCert, rootCert, pemEncodedChain[1], pemEncodedChain[2]);
-    verifyResponseSignature(response, signature, pemEncodedChain[0], null);
+    verifyResponseSignature(response, signature, pemEncodedChain[0], null, hashAlg);
   }
 
   for (var j = 0; j < apkDigest.length; j++) {
     for (var i = 0; i < apkCertDigest.length; i++) {
       try {
-        verifyPayload(jwsPayload, response, requestID, signature, apkDigest[j], apkCertDigest[i]);
+        verifyPayload(jwsPayload, response, requestID, signature, apkDigest[j], apkCertDigest[i], version);
       } catch (err) {
         if (err.message !== 'verifyPayload failed: wrong signing certificate hash' &&
           err.message !== 'verifyPayload failed: wrong apk hash') {
-          throw new Error('verifyPayload failed: apk hash or signing cert hash mismatch');
+          throw new Error(err.message);
         }
       }
     }
   }
 
-  verifySignature(jws, googleCert);
-  await verifyAuthenticity(jws, googleApiKey);
+  if (!verifySignature(jws, googleCert)) {
+    throw new Error('verifySignature failed');
+  }
+
+  for (var i = 0; i < googleApiKey.length; i++) {
```

Comment:
always use `let` over `var`. If you want know the reason try open the node repl and try `for(var i=0; i<10; i++){console.log(9)}` then `for(let k=0; k<10; k++){console.log(9)}` then try to print `i` and `k`

Reply:
Wow, js is really weird... thanks for pointing this out!